sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample4_StronglyTypingARecognizedForm.md

```diff
@@ -0,0 +1,79 @@
+# Strongly-typing a recognized form
+
+This sample demonstrates how to use the fields in your recognized forms to create an object with strongly-typed fields. The pre-trained receipt model will be used to illustrate this sample, but note that a similar approach can be used for any custom form as long as you properly update the fields' names and types.
+
+To get started you'll need a Cognitive Services resource or a Form Recognizer resource.  See [README][README] for prerequisites and instructions.
+
+## Creating a `FormRecognizerClient`
+
+To create a new `FormRecognizerClient` you need the endpoint and credentials from your resource. In the sample below you'll use a Form Recognizer API key credential by creating an `AzureKeyCredential` object, that if needed, will allow you to update the API key without creating a new client.
+
+You can set `endpoint` and `apiKey` based on an environment variable, a configuration setting, or any way that works for your application.
+
+```C# Snippet:CreateFormRecognizerClient
+string endpoint = "<endpoint>";
+string apiKey = "<apiKey>";
+var credential = new AzureKeyCredential(apiKey);
+var client = new FormRecognizerClient(new Uri(endpoint), credential);
+```
+
+## Strongly-typing a receipt
+
+`FormField<T>` is a helper class used in this sample as a strongly-typed version of `FormField`. They have the same properties, except for `Value`, which returns a `T` instead of a `FieldValue`.
+
+The known receipt fields returned by the service, such as `MerchantName` and `TransactionDate`, will be strongly-typed and put together into another helper class called `Receipt`.
+
+```C# Snippet:FormRecognizerSampleStronglyTypingARecognizedForm
+RecognizedFormCollection recognizedForms = await client.StartRecognizeReceiptsFromUri(new Uri(receiptUri)).WaitForCompletionAsync();
+
+foreach (RecognizedForm recognizedForm in recognizedForms)
+{
+    Receipt receipt = new Receipt(recognizedForm);
```

Comment:
You link to the class file for this below, but it would be great to see that link before the sample so that we know that this is a user-defined type.

Reply:
Should this more specifically be USReceipt?

Reply:
I didn't think that much about the name. I wonder if we'll have any trouble with it, though. If we reintroduce the `USReceipt` type in the main library in the future, it would break the code of customers who followed the approach in this sample. We didn't have a `Receipt` type (since we called it `RecognizedReceipt`), but maybe the name is too generic and may cause issues. Thoughts?

Reply:
I think since we are adding this class in the samples section, we aren't bound to support non-breaking changes to this class. Also, since we no longer have a `receiptLocale` it becomes necessary to correctly classify a generic receipt to its specific type. For future advancements, any other types could go as their own separate classes.

Reply:
> we aren't bound to support non-breaking changes to this class

Agreed.

> Also, since we no longer have a `receiptLocale` it becomes necessary to correctly classify a generic receipt to its specific type.

I understand your reasoning, but I think we have different interpretations of what this sample is trying to illustrate (you may have already figured that out given my other replies, so apologies for being redundant). We're showing how to strongly-type a custom form and using a pre-trained receipt model as an example, so we don't intend to extend this sample even if other types of receipts are included.